### PR TITLE
Rework of UI translation System

### LINKED ISSUE #785: B-Wing Hera  Quick Build includes the wrong title
### Expected Behavior

The Quick Build for Hera Syndulla in a B-Wing, should include the Epic version of "B6 Blade Wing Prototype". (crazy though, right?)

### Actual Behavior

Instead it loads the standard title.

### Language

-

### PR DESCRIPTION
**Pull request by Patschke**
The current "System" to translate the UI is ugly as hell. The translation file has to access stuff using CSS selectors. The rework will instead mark all translateable UI elements properly, leaving just a english->foreign language in the translation files. Hopefully this cleanup will also speed up language changes a bit. 

There is also a log output informing about missing translations, that allows people that are less familiar with coding to add in new UI translations (previously one needed to understand html/css). It can be enabled by setting `SHOW_DEBUG_OUT_MISSING_TRANSLATIONS = true;` in the developer console. 

This PR is hopefully ready to be merged. Testing is required for all backend-related stuff: This includes checking the UI for squad saving/loading/etc as well as saved language preferences (e.g. does the language switch to the saved preference after logging in?). 

This PR also closes #785

### CODE REVIEW

File: coffeescripts/content/cards-en.coffee [hunk 1]
```diff
@@ -83,89 +88,42 @@ exportObj.translations.English =
...
-        '.game-type-selector option[value="epic"]': 'Epic'
-        '.game-type-selector option[value="quickbuild"]': 'Quick Build'
-        # Card browser
-        '.xwing-card-browser option[value="name"]': 'Name'
-        '.xwing-card-browser option[value="source"]': 'Source'
-        '.xwing-card-browser option[value="type-by-points"]': 'Type (by Points)'
-        '.xwing-card-browser option[value="type-by-name"]': 'Type (by Name)'
-        '.xwing-card-browser .translate.select-a-card': 'Select a card from the list at the left.'
-        '.xwing-card-browser .translate.sort-cards-by': 'Sort cards by'
-        # Info well
```
Comment (raithos) [inline coffeescripts/content/cards-en.coffee:116]: A little concerned about removing all of the info well translations as well as the buttons and obstacles. Are we replicating them somewhere else that I'm not aware of?

**Comment by raithos** [comment]
overall this is really great work. I really like auto-selecting the language based on browser preferences. I have minor comments

File: coffeescripts/content/cards-en.coffee (hunk 1 above)
Comment (Patschke) [reply coffeescripts/content/cards-en.coffee:116]: Everything that has a `translate` class attribute will also have a `defaultText` attribute. This default text will then be what is translateable in the language file. I hope that I'll replicate all those entries correctly.

**Comment by raithos** [comment]
do you need me to merge this now for any reason or can I wait until you're done with commits?

**Comment by Patschke** [comment]
I don't need you to merge this now, I should be able to complete and test most stuff local. Only backend-related stuff is harder to test, since I still was too lazy to fire up my own backend. The online backend does have cross-origin rules that prevent access from my local system.

**Comment by Patschke** [comment]
Basically there are two ways UI translation takes place: 
  * On language change and initial loading, all texts of html nodes with class `translated` are set to the translated texts. For this, the static html needs to contain the class and a `defaultText` attribute that provides the information which text belongs inside. 
   This requires just one loop over the created html to find all nodes with `translated` class, while finding the tranlated texts are basically dict accesses. The old CSS selector scheme where each translation basically provided it's own description of where to be inserted required a search through all html nodes for each individual translation. 
  * Dynamically generated UI (e.g. the delete-pilot button for each ship) is translated on creation, using the `exportObj.translate` method. 

Drawbacks I'm aware of: 
  * If you dynamically create "static" html (which almost all our UI is, as the jade files are rather small), you need to invoke the translation run for initial loading manually. This is done in all setupUI methods by calling `exportObj.translateUIElements`. The newly generated html can be passed as context, to prevent re-translations of the whole page. I'm fine with that. 
  * UI text that is within attributes can't be translated on language change. This affects e.g. placeholders or the text of the current choice in select dropdowns.

**Comment by Patschke** [comment]
I hate when this happens: I thought I messed something up on the xws loading stuff, but actually there is just a bug in the live version. After you load some xws, a `Uncaught TypeError: b.xws_import_modal is undefined` is issued. Obviously I couldn't find where I introduced this bug, as it was already there... going to fix it...

**Comment by Patschke** [comment]
Hopefully ready to be merged!

The next step would be to actually use all the new functionality and provide a complete UI translation for german, to have a somewhat proper template other languages could use.

**Comment by raithos** [comment]
Something you did seemed to have broken logins. I've reverted it on live for now. I'll upload a copy to /test for us to troubleshoot

**Comment by Patschke** [comment]
I was afraid that this could happen - in the end it is after all a somewhat major refactor, and login was completely untested...

**Comment by raithos** [comment]
yeah im not even sure if theres a problem anymore, it like tweeked out but now when i try to reproduce it on /test it doesnt happen anymore

**Comment by Patschke** [comment]
There is a error popping up in the log for me. 
```
Uncaught TypeError: a.collection.onLanguageChange is not a function
    setupEventHandlers xwing.src.coffee:3333
    jQuery 7
    setupHandlers xwingcontent.min.js:20
    a xwingcontent.min.js:20
    loadCollection xwing.src.coffee:993
    jQuery 7
    loadCollection xwing.src.coffee:991
    setupHandlers xwing.src.coffee:914
    jQuery 7
    maybeAuthenticationChanged xwing.src.coffee:381
    __bind xwing.src.coffee:1
    success xwing.src.coffee:374
    jQuery 6
    authenticate xwing.src.coffee:367
    setupHandlers xwing.src.coffee:928
    jQuery 2
```
So I guess there is something to be fixed ;-)
Reloading after login does "fix" parts of it (I am able to access saved squads, but not my collection).

**Comment by raithos** [comment]
There also seems to be issues picking pilots/upgrades after logging in as well. probably related to the collection issues. And on firefox you cant select anything after switching

**Comment by Patschke** [comment]
I guess fixing the first bug first makes sense. It's possible that the collection does not get loaded properly. As all drop-downs access it to show not-owned stuff in red, they will also fail to load.

**Comment by Patschke** [comment]
Not sure how I entierely missed that, but the manifest does contain html code. I did remove the `onLanguageChanged` methods (as all UI would be translated globally), which was not only called within `manifest.coffee`, but also from `xwing.coffee`. I'll have to adapt that to the new system, for now I just created a PR commenting out the call to the non-existing method from `xwing.coffee` (which is a weired call anyway - the collection did subscribe to the trigger and would call it's `onLanguageChanged` method itself (a call I did remove), so there should've never been any reason to call it additionally from the trigger of every single browser. In the end, it was probably called nine times on each language change (once per builder plus once from it's own trigger)...

**Comment by raithos** [comment]
Yeah the whole onlanguagechanged logic pathway didnt make sense to me but when i tried to streamline it it caused issues where the english would not show up when there weren't other translations, even if logically it should have worked. I'll send that PR to test and lets see if anything else breaks

**Comment by Patschke** [comment]
I think the quickfix PR is able to fix the critical bug that was present here. 

As I did not notice that there is html in `manifest.coffee`, I'll need to work on that as well. As that collection-management stuff will only be working if one is logged in, testing for this is also easier on the /test live build than on local builds.

**Pull request merged**